include/aspect/simulator.h

```diff
@@ -2014,6 +2014,7 @@ namespace aspect
       template <int dimension, int velocity_degree>
       friend class StokesMatrixFreeHandlerImplementation;
       friend struct Parameters<dim>;
+      friend class TimeHandler<dim>;
```

Comment:
Eww, another friend class? ;-) Can we avoid this if you move the `compute_time_step` function into the `TimeHandler` as well? This seems to be the only thing needing the friend access, and it is the logical place for it anyway. It may need some changes inside the function, but with `SimulatorAccess` everything in `compute_time_step` should be possible.